sequence_processing_pipeline/Pipeline.py

```diff
@@ -12,11 +12,79 @@
 import sample_sheet
 import pandas as pd
 from collections import defaultdict
+from datetime import datetime
+import pytz
+from xml.etree import ElementTree as ET
 
 
 logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
 
 
+class InstrumentUtils():
+    types = {'A': 'NovaSeq 6000', 'D': 'HiSeq 2500', 'FS': 'iSeq',
+             'K': 'HiSeq 4000', 'LH': 'NovaSeq X Plus', 'M': 'MiSeq',
+             'MN': 'MiniSeq',
+             # SN – RapidRun which is HiSeq 2500
+             'SN': 'RapidRun'}
+
+    @staticmethod
+    def get_instrument_type(run_id):
+        tmp = run_id.split('_')
+
+        # remove digits from substring to get instrument code.
+        code = ''.join([i for i in tmp[1] if not i.isdigit()])
+        try:
+            return InstrumentUtils.types[code]
+        except KeyError:
+            raise ValueError(f"Instrument code '{code}' is of unknown type")
+
+    @staticmethod
+    def get_instrument_id(run_id):
+        tmp = run_id.split('_')
+
+        # remove letters from substring to get instrument id.
+        return ''.join([i for i in tmp[1] if i.isdigit()])
+
+    @staticmethod
+    def get_date(run_id):
+        ds = run_id.split('_')[0]
+
+        if len(ds) == 6:
```

Comment:
why not use datetime.strptime(date_string, format) ?

Reply:
That's a good point. I'll check the docs and see if date_string can handle regex or variable patterns.